### ISSUE TITLE
Add easy way to fully disable Push or Pull methods

### ISSUE THREAD

**Issue by dkotter** [10up/distributor#815]
**Is your enhancement related to a problem? Please describe.**
We support both Pushing and Pulling content out of the box but often sites will implement an editorial process that only uses one or the other. It's not bad to have both methods available but if one of those methods is never used or in a stricter editorial site where they want to ensure one method is never used, would be nice to fully disable that unused method.

**Describe the solution you'd like**
Ideally there would be a simple way to disable either Push or Pull. There are ways to do this now but they feel a little hacky and it's not really clear how to go about doing it.

I imagine two new filters (`dt_is_pull_disabled`, `dt_is_push_disabled` or similar) that return `false` by default but if you add your own custom code to return `true` instead, those methods are completely disabled.

**Comment by jeffpaul** [comment]
Perhaps could close this via https://10up.github.io/distributor/dt_pull_capabilities.html && https://10up.github.io/distributor/dt_push_capabilities.html?

**Comment by peterwilsoncc** [comment]
WordPress's capability checking includes `do_not_allow` as a magic phrase to prevent users from doing an action. To disable pull permissions with the filters mentioned above can be done with:

```php
add_filter( 'dt_pull_capabilities', function() { return 'do_not_allow'; } );
```

The above results in the check `current_user_can( 'do_not_allow' )` which returns false for all users, including super admins. I think it's safe to close this as unplanned as the feature exists. Perhaps we can add this as a documentation enhancement.

**Comment by jeffpaul** [comment]
Yeah, somehow adding into our docs would probably be good, perhaps in the Snippets section?
